src/middlewares/validation.ts

```diff
@@ -146,4 +149,44 @@ const verifyUserCredentials = async (
     }
 };
 
-export { validation, isUserExist, isAccountVerified, verifyUserCredentials, isUsersExist };
\ No newline at end of file
+const isProductExist = async(req: any, res: Response, next: NextFunction) => {
+    try {
+        const shop = await productRepositories.findItemByAttributes(Shops,"userId", req.user.id);
+        if(!shop){
+            return res.status(httpStatus.NOT_FOUND).json({ status: httpStatus.NOT_FOUND, message: "Not shop found." });
+        }
+        const isProductAvailable = await productRepositories.findByModelAndAttributes(Products,"name","shopId", req.body.name,shop.id);
+        if(isProductAvailable){
+            return res.status(httpStatus.BAD_REQUEST).json({ status: httpStatus.BAD_REQUEST, message: "Product already exists. Please update the stock." });
+        }
+        req.shop = shop;
+        next();
+    } catch (error) {
+        return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({ status: httpStatus.INTERNAL_SERVER_ERROR, message: error.message });
+    }
+}
+
+
+const isShopExist = async (req: any, res: Response, next: NextFunction) =>{
+    try {
+        const shop = await productRepositories.findItemByAttributes(Shops, "userId",req.user.id)

```

Comment:
Let us be specific in the creation of function instead of **`findItemByAttributes`** let be  **`findShopByAttributes`**